src/kafka/producer.js

```diff
@@ -12,9 +12,20 @@ module.exports = async (record, topic = 'test_123') => {
     });
     await producer.connect();
 
+    let payload;
+    if (encodedRecord) {
+        payload = encodedRecord;
+    } else {
+        payload = JSON.stringify(record);
+    }
+
     await producer.send({
         topic: topic,
-        messages: [{ value: JSON.stringify(record) }]
+        messages: [{
+            // TODO: keys don't work yet.
+            key: record["id"],
```

Comment:
I think that if we change `record["id"]` to `record.id,` it should work

Reply:
Both ways work (as long as the key “id” exists!) — What I meant with that comment is it doesn’t work in general for user defined keys. We would have to tackle that in a separate PR where maybe the parsers return the (possibly null) key and the record instead of just the record